src/megaapi_impl.cpp

```diff
@@ -24975,9 +24985,12 @@ void MegaFolderUploadController::start(MegaNode*)
         return;
     }
 
-    auto localpath = LocalPath::fromPath(transfer->getPath(), *client->fsaccess);
-    scanFolder(parent->getHandle(), parent->getHandle(), localpath, folderName);
-    createFolder();
+    mThread = std::thread ([this]() {
+        auto localpath = LocalPath::fromPath(transfer->getPath(), *client->fsaccess);
+        scanFolder(transfer->getParentHandle(), transfer->getParentHandle(), localpath, transfer->getFileName());
+        createFolder();
```

Comment:
I think it would be easier to read if the `wait()` after createFolder and the call to `uploadFiles()` were all in the lambda for the thread.